### DISCUSSION TITLE
Run webhook permanently and on startup

### DISCUSSION THREAD

**Discussion by marco910** [adnanh/webhook#562] [General]
Hi community,

How can I run "webhook" permanently and start it on system startup? Is using PM2 a good solution?

**Comment by moorereason** [comment]
I don't use PM2, but [I hear it works](https://github.com/adnanh/webhook/issues/430).  My systems use systemd.

**Comment by marco910** [comment]
@moorereason thanks for your reply!

One more dependencie is always good :) Do you have an example of your setup or how you started the job forever?

**Comment by moorereason** [comment]
You can put webhook in `/opt/webhook` and create the following `/opt/webhook/webhook.service` for systemd.

```
[Unit]
Description=Webhook Service
After=network.target

[Service]
ExecStart=/opt/webhook/webhook -hooks=/opt/webhook/hooks.json -hotreload=false -ip= -port=9000 -secure=false -cert=cert.pem -key=key.pem -verbose=true -debug=false
WorkingDirectory=/opt/webhook
KillMode=process
Restart=on-failure

[Install]
WantedBy=multi-user.target
```

**Comment by marco910** [comment]
There is already a service for webhook, which is inactive. But I can't start that and port 9000 is already in use (is think by this service).

**Reply by marco910** [reply]
> There is already a service for webhook, which is inactive. But I can't start that and port 9000 is already in use (is think by this service).

I fixed that by disabling the default webhook-service in systemd with `sudo systemctl disable webhook`. After that, I create a new service for systemd as @moorereason explains and linked that to `/etc/systemd/system` by using `sudo ln -s`.
At the end i restarted systemd with the command `sudo systemctl daemon-reload`, enabled the new service `sudo systemctl enable mywebhookservice.service` and started the service with `sudo systemctl start mywebhookservice.service`.

Thanks for your help, @moorereason !
